### PR TITLE
ENG-14522-voltsql improvement

### PR DESCRIPTION
**Pull request by Chaozz**
This PR is a mixed fix for several tickets: ENG-14522, ENG-14525, ENG-14526, ENG-14527.

List of what it does:

- give a better error message for missing dependencies, together with the scripts to install them.
- will tell the user to install pip if it is not installed
- now multiline-mode by default is OFF.
- store the command history at `~/.voltsql_history`. It can be used for retyping previous commands.
- add prioritization to keywords, it is calculated based on its occurrence in `~/.voltsql_history`.
- clean up and update the `volt_literals.json`
- add a `help` command in voltsql which will print the readme 
- update the readme

### CODE REVIEW

File: lib/python/voltsql/voltsql/voltliterals/volt_literals.json [hunk 1]
```diff
@@ -408,17 +328,12 @@
         "AUTHORIZATION",
         "BINARY",
         "COLLATION",
-        "CONCURRENTLY",
         "CROSS",
-        "CURRENT_SCHEMA",
-        "FREEZE",
-        "FULL",
```
Comment (yzhang1991) [inline lib/python/voltsql/voltsql/voltliterals/volt_literals.json]: [INNER | {LEFT | RIGHT | FULL } [OUTER]] JOIN.

File: lib/python/voltsql/voltsql/voltliterals/volt_literals.json [hunk 2]
```diff
@@ -408,17 +328,12 @@
...
         "ILIKE",
         "INNER",
         "IS",
         "ISNULL",
         "JOIN",
-        "LEFT",
```
Comment (yzhang1991) [inline lib/python/voltsql/voltsql/voltliterals/volt_literals.json]: [INNER | {LEFT | RIGHT | FULL } [OUTER]] JOIN.

File: lib/python/voltsql/voltsql/voltliterals/volt_literals.json [hunk 3]
```diff
@@ -408,17 +328,12 @@
...
         "LIKE",
```
Comment (yzhang1991) [inline lib/python/voltsql/voltsql/voltliterals/volt_literals.json]: Is there a "STARTS" and "WITH"? Make sure it has those.

File: lib/python/voltsql/voltsql/voltsql.py [hunk 4]
```diff
@@ -43,21 +43,48 @@
...
+]
+
+try:
+    require(dependencies)
+except (DistributionNotFound, VersionConflict) as error:
+    voltsql_root_dir = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), os.pardir))
+    try:
+        import pip
+    except ImportError:
+        print("You need to install pip first. Then you can install the dependencies using the following command: \n"
```
Comment (yzhang1991) [inline lib/python/voltsql/voltsql/voltsql.py]: @ajgent Is this OK?

File: lib/python/voltsql/README.md [hunk 5]
```diff
@@ -6,39 +6,80 @@ Install Requirements
 
 - We require Python 2.6+, we don't support Python 3 currently.
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply lib/python/voltsql/README.md]: Andrew's PR will fix this...

File: lib/python/voltsql/voltsql/voltliterals/volt_literals.json [hunk 6]
```diff
@@ -60,123 +49,64 @@
...
-        "RAW": [],
-        "REFRESH MATERIALIZED VIEW": [],
-        "RESOURCE": [],
-        "RETURNS": [],
-        "REVOKE": [],
         "RIGHT": [],
         "ROLLBACK": [],
         "ROW": [],
-        "ROWID": [],
-        "ROWNUM": [],
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply lib/python/voltsql/voltsql/voltliterals/volt_literals.json]: I have added the row_number().

File: lib/python/voltsql/voltsql/voltliterals/volt_literals.json [hunk 7]
```diff
@@ -93,6 +96,7 @@
         "SELECT": [],
         "SET": [],
         "SHOW": [],
+        "STARTS WITH": [],
```
Comment (yzhang1991) [inline lib/python/voltsql/voltsql/voltliterals/volt_literals.json]: Add a stand-alone WITH, and RECURSIVE.

Comment (Chaozz) [reply lib/python/voltsql/voltsql/voltliterals/volt_literals.json]: done

**Pull request merged**